lib/DB/Schema/ResultSet/Course.pm

```diff
@@ -142,7 +142,10 @@ sub addCourse ($self, %args) {
 	}
 	$params->{course_settings} = {};
 
-	# Check the parameters.
+	# Add the visible default if not there.  Note: although in the database this value is
+	# already set, it appears that in the create method below, the default value is
+	# not created.  Perhaps rethink this a bit.
+	$params->{visible} = 1 unless defined $params->{visible};
 	my $new_course = $self->create($params);
```

Comment:
This definitely needs to be rethought out.  The default value for the column in the database is set even if the 'visible' key is not in the params hash.  You are only adding this here so that it will be in the returned data.  Not because it has any effect at all on the creation of the course in the database.

What is really your intent with taking the return value of the `create` method, and using it for the return value of the `addCourse` method?  The only thing in that return value that has any real meaning is the `course_id`.

Reply:
I think that the right way to handle this is to add `retrieve_on_insert => 1` to the visible column definition in the schema result source.  Then you can delete the line 148 (and preceding comments), and you will get the `visible` value back even if it is not in the params.